cosmos/converter.py

```diff
@@ -229,8 +229,8 @@ def __init__(
 
         validate_changed_config_paths(execution_config, project_config, render_config)
 
-        env_vars = copy.deepcopy(project_config.env_vars or operator_args.get("env"))
-        dbt_vars = copy.deepcopy(project_config.dbt_vars or operator_args.get("vars"))
+        env_vars = project_config.env_vars or operator_args.get("env")
```

Comment:
@pankajastro could you add the DAG from the description as an example DAG, so we can have some way to make sure the desired use-case continues working after future chages?

Reply:
Added test. thanks!